lib/gateway/Shard.js

```diff
@@ -628,15 +640,21 @@ class Shard extends EventEmitter {
                         message.reactions = {};
                     }
                 }
+                if(!message) {
+                    message = {
+                        id: packet.d.message_id,
+                        channel: channel || {id: packet.d.channel_id}
+                    };
+                    if(packet.d.guild_id) {
+                        message.guildID = packet.d.guild_id;
```

Comment:
```suggestion
                        message.guildID = packet.d.guild_id;
                        message.channel.guild = {id: packet.d.guild_id};
```

Reply:
Not sure if this is necessary but meh

Reply:
If the channel is cached `message.channel.guild` should still be present. This would overwrite it?

Reply:
Ah yeah it would.. mb
Probably add something to check if guild doesn't exist before doing that then

Reply:
Added for all 4 events